### PR TITLE
chore: refactor federation factory

### PR DESCRIPTION
**Pull request by Aenimus**
## TODO

- [ ] Tests or benchmark included
- [ ] Documentation is changed or added on [https://app.gitbook.com/](https://app.gitbook.com/)
- [ ] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)

### CODE REVIEW

File: studio/package.json [hunk 1]
```diff
@@ -111,7 +111,7 @@
     "tailwindcss-animate": "^1.0.7",
     "tiptap-markdown": "^0.8.8",
     "use-deep-compare-effect": "^1.8.1",
-    "zod": "^3.22.3"
```
Comment (StarpTech) [inline studio/package.json:114]: If a different zod version produces bugs, let's fix the version.

File: composition/src/normalization/normalization-factory.ts [hunk 2]
```diff
@@ -118,6 +117,7 @@ import {
   FIELDS,
   FROM,
   KEY,
+  N_A,
```
Comment (StarpTech) [inline composition/src/normalization/normalization-factory.ts:120]: What is N_A?

File: composition/src/normalization/normalization-factory.ts [hunk 3]
```diff
@@ -530,13 +520,15 @@ export class NormalizationFactory {
...
       this.parentExtensionDataByTypeName,
       node,
       this.errors,
       this.directiveDefinitionByDirectiveName,
       this.handledRepeatedDirectivesByHostPath,
       isEntity,
+      isRootType,
+      this.subgraphName,
     );
     // TODO re-assess this line
```
Comment (StarpTech) [inline composition/src/normalization/normalization-factory.ts:533]: Still a todo?

File: composition/src/schema-building/ast.ts [hunk 4]
```diff
@@ -217,7 +232,7 @@ export type MutableNonNullTypeNode = {
 };
 
 // TODO errors
```
Comment (StarpTech) [inline composition/src/schema-building/ast.ts:234]: Whats the exact todo here?

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: composition/src/schema-building/ast.ts (hunk 4 above)
Comment (Aenimus) [reply composition/src/schema-building/ast.ts:234]: I forgot to replace the error. Thanks for pointing it out—done.

File: composition/src/normalization/normalization-factory.ts (hunk 3 above)
Comment (Aenimus) [reply composition/src/normalization/normalization-factory.ts:533]: Yes. for now. It's to remind me to check the logic here shouldn't change after implementing entity interfaces.

**Review by StarpTech — APPROVED** [review]
LGTM

**Pull request merged**